workitem/workitem.go

```diff
@@ -53,3 +57,66 @@ func (wi WorkItem) GetLastModified() time.Time {
 	log.Debug(nil, map[string]interface{}{"wi_id": wi.ID}, "Last modified value: %v", lastModified)
 	return *lastModified
 }
+
+// ChangeSet derives a changeset between this workitem and a given workitem.
+func (wi WorkItem) ChangeSet(older convert.ChangeDetector) ([]convert.Change, error) {
+	olderWorkItem, ok := older.(WorkItem)
+	if !ok {
+		return nil, errors.New("Other entity is not a WorkItem: " + reflect.TypeOf(older).String())
+	}
+	if wi.ID != olderWorkItem.ID {
+		return nil, errors.New("Other entity has not the same ID: " + olderWorkItem.ID.String())
+	}
+	changes := []convert.Change{}
+	// CAUTION: we're only supporting changes to the system.state and to the
+	// board position relationship for now. If we need to support more
+	// attribute changes, this has to be added here. This will be likely
+	// necessary when adding new Actions.
+	// compare system.state
+	if wi.Fields["system.state"] != olderWorkItem.Fields["system.state"] {
```

Comment:
Please use `workitem.SystemState` when you want to refer to `system.state`.

Reply:
Outdated. Already fixed.